src/sources/util/net/mod.rs

```diff
@@ -67,19 +89,53 @@ impl From<SocketAddr> for SocketListenAddr {
     }
 }
 
-fn parse_systemd_fd<'de, D>(des: D) -> Result<usize, D::Error>
-where
-    D: Deserializer<'de>,
-{
-    let s: &'de str = Deserialize::deserialize(des)?;
-    match s {
-        "systemd" => Ok(0),
-        s if s.starts_with("systemd#") => s[8..]
-            .parse::<usize>()
-            .map_err(de::Error::custom)?
-            .checked_sub(1)
-            .ok_or_else(|| de::Error::custom("systemd indices start from 1, found 0")),
-        _ => Err(de::Error::custom("must start with \"systemd\"")),
+impl From<usize> for SocketListenAddr {
+    fn from(fd: usize) -> Self {
+        Self::SystemdFd(fd)
+    }
+}
+
+impl TryFrom<String> for SocketListenAddr {
+    type Error = SocketListenAddrParseError;
+
+    fn try_from(input: String) -> Result<Self, Self::Error> {
+        // first attempt to parse the string into a SocketAddr directly
+        match input.to_socket_addrs() {
```

Comment:
`ToSocketAddrs::to_socket_addrs` is not what we want. That will actually attempt to resolve something as a hostname if it doesn't validate as an IP address.

You should just use `SocketAddr::from_str`.

Reply:
fixed in [f21cddc](https://github.com/vectordotdev/vector/pull/16027/commits/f21cddc98823fa6d4d175dc0689a33fa80cbdd1e)